addons/bus/static/src/im_status_service.js

```diff
@@ -41,9 +42,16 @@ export const imStatusService = {
             }
         };
 
-        bus_service.addEventListener("connect", () => {
-            browser.setTimeout(updateBusPresence, FIRST_UPDATE_DELAY);
-        });
+        bus_service.addEventListener(
+            "subscribed",
+            () => {
+                if (!initialSubscriptionDone) {
+                    initialSubscriptionDone = true;
+                    updateBusPresence();
+                }
+            },
+            { once: true }
```

Comment:
What happens in case of disconnecting/reconnecting?

Reply:
If presence message is sent during a disconnection, message will be queued and sent on the next socket open.